docs/modeling-your-data/modeling-your-data-with-dbt/dbt-advanced-usage/dbt-incremental-logic/index.md

```diff
@@ -68,40 +90,138 @@ In all states the `upper_limit` is limited by the `snowplow__backfill_limit_days
 
+```mermaid
+%%{init: { 'gantt': {'barHeight': 100, 'fontSize': 30, 'sectionFontSize': 30, 'leftPadding': 200, 'numberSectionStyles':3, 'displayMode': 'compact'}}}%%
+gantt
+    dateFormat  YYYY-MM-DD
+    axisFormat  
+    tickInterval 10day
+    section  
+    snowplow__start_date :crit, m1, 2020-01-01, 2m
+
+    section Unprocessed
+    Future Runs (new model) :crit, c, after a1 , 2d
+
+    section Processing
+    ⬅snowplow__backfill_limit_days➡ :active, a1, 2020-01-01, 1d
+
+    section Processed
+    Other models :c, 2020-01-01, 3d
+
+```
+
 ### State 3: Models out of sync
 
-`min_last_success < max_last_success` and therefore the tagged models are out of sync, for example due to a particular model failing to execute successfully during the previous run. The package will attempt to sync all models.
+If the `min_last_success` is less than the `max_last_success` it means the tagged models are out of sync, for example due to a particular model failing to execute successfully during the previous run or as part of catching up on a new model. The package will attempt to sync all models as far as it can.
 
 **`lower_limit`**: `min_last_success - snowplow__lookback_window_hours`  
 **`upper_limit`**: `least(max_last_success, min_last_success + snowplow__backfill_limit_days)`
 
+```mermaid
+%%{init: { 'gantt': {'barHeight': 100, 'fontSize': 30, 'sectionFontSize': 30, 'leftPadding': 200, 'numberSectionStyles':3, 'displayMode': 'compact'}}}%%
+gantt
+    dateFormat  YYYY-MM-DD
+    axisFormat  
+    tickInterval 10day
+    section  
+    min_last_success :crit, m1, 2020-01-02, 4m
+
+    section Unprocessed
+    Future Runs (out of sync model) :crit, c, after a2, 2d
+
+    section Processing
+    ⬅snowplow__lookback_window_hours➡ :active, a1, 2020-01-01 18:00:00, 6h
+    ⬅snowplow__backfill_limit_days➡ :active, a2, after b1, 1d
+
+    section Processed
+    Out of sync model :b1, 2020-01-01, 1d
+    Other models :b2, 2020-01-01, 4d
+
+```
 ### State 4: Standard run
 
-If none of the above criteria are met, then we consider it a 'standard run' and we carry on from the last processed event.
+If none of the above criteria are met, then we consider it a 'standard run' where all models are in sync and we carry on from the last processed event.
 
 **`lower_limit`**: `max_last_success - snowplow__lookback_window_hours`  
 **`upper_limit`**: `least(current_tstamp, max_last_success + snowplow__backfill_limit_days)`
 
 
+```mermaid
+%%{init: { 'gantt': {'barHeight': 100, 'fontSize': 30, 'sectionFontSize': 30, 'leftPadding': 200, 'numberSectionStyles':3, 'displayMode': 'compact'}}}%%
+gantt
+    dateFormat  YYYY-MM-DD
+    axisFormat  
+    tickInterval 10day
+    section  
+    max_last_success :crit, m1, 2020-01-04, 4m
+
+    section Unprocessed
+    Future Runs :crit, c, after a2 , 1d
+
+    section Processing
+    ⬅snowplow__lookback_window_hours➡ :active, a1, 2020-01-03 18:00:00, 6h
+    ⬅snowplow__backfill_limit_days➡ :active, a2, after b1, 1d
```

Comment:
Could the snowplow__backfill_limit_days go to the right of the processing block as it is the upper limit that this variable changes?

Reply:
I might just remove the arrows, it works for the first one where it's in the block but outside it does confuse things a little